### PR TITLE
#1777 - Process MSFAA Cancellation Notification e2e tests

### PR DESCRIPTION
**Pull request by sh16011993**
When the full-time or part-time cancellation record files are received, the corresponding MSFAA records are cancelled and notification is persisted in the database.

As a part of this PR, e2e tests are written to validate the notifications when the MSFAA record gets cancelled.

### CODE REVIEW

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (ann-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:280]: shouldn't we also check, if the record status was updated or not?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (ann-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:283]: can't we use Typeform query

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (ann-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:171]: https://github.com/bcgov/SIMS/pull/2069/files#r1248261900

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (ann-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:177]: https://github.com/bcgov/SIMS/pull/2069/files#r1248264087

**Review by ann-aot — COMMENTED** [review]
added some comments

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:293]: Can we also check of the notification type is the expected one?

Comment (andrepestana-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:293]: I think we don't need to do a `expect` for the `notificationMessageType` because it's  been selected by it if I understood.

Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:293]: @andrewsignori-aot I agree with @andrepestana-aot . Since it is already a part of the `where` clause, won't it be redundant?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:242]: Can we please have it adjusted as below?
"Should cancel a pending MSFAA record and save a cancellation notification when an MSFAA cancellation record is received in the response file."

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:293]: Yes, my bad, I missed it. Resolving the comment.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:283]: I tried to use it at first but then struggled in putting up the where clause using the typeorm query.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:177]: https://github.com/bcgov/SIMS/pull/2069/files#r1248311140

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (dheepak-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:293]: @andrewsignori-aot Adding to your comment, the userId of the notification can be expected to be studentUserId

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:98]: Can we have it adjusted as below?
"Should process an MSFAA response with confirmations and cancellations, and save a notification message when the file is received as expected.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrepestana-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:283]: Example:
```js
   {
      select: {
        id: true,
        overawardNotes: { noteType: true, description: true },
      },
      relations: { overawardNotes: true },
      where: { id: overawardResponse.body.id },
    }
```

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:252]: createFakeMSFAANumber created with default state pending right?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:177]: Can you try the below one?
```ts
db.notification.find({
  select: {
    id: true,
    dateSent: true,
  },
  where: {
    notificationMessage: {
      id: notificationMessageType,
    },
    user: {
      id: studentUserId,
    },
  },
});
```

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments.

**Comment by dheepak-aot** [comment]
Good job! Just have one additional minor comments. Other DEVS have pretty much covered everything.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:177]: Thank You @andrewsignori-aot

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:171]: The record status is already being validated a few lines below:
 
`expect(cancelledMSFAA.cancelledDate).toBe("2021-11-24");`      
`expect(cancelledMSFAA.newIssuingProvince).toBe("ON");`

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:280]: Added it. Thanks @ann-aot

**Review by andrepestana-aot — COMMENTED** [review]
Great job! Please at the other devs comments. Don't wait for my approval 😉

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:283]: Thanks @andrepestana-aot

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:252]: Yes. Removed it @dheepak-aot . Thanks 👍

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:309]: Thanks for adding this verification.
@ann-aot @guru-aot @dheepak-aot I believe that we can check the dynamic payload saved also as much as possible as part of the E2Es.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts]: Was this suggestion accepted?

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, look good 👍

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (ann-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts:141]: do we need `id`?

**Review by ann-aot — APPROVED** [review]
LGTM 👍 just a minor comment

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts
Comment (sh16011993) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-response-integration.scheduler.e2e-spec.ts]: Thanks @ann-aot . Removed it.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes 👍

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts
Comment (guru-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts:309]: Yes, this is good

**Review by guru-aot — COMMENTED** [review]
LGTM, nice work @sh16011993

**Pull request merged**
